app/app.go

```diff
@@ -744,6 +785,28 @@ func (app *App) SupportedVersions() []uint64 {
 	return app.mm.SupportedVersions()
 }
 
+// versionedKeys returns the keys for the kv stores for a given app version
+func (app *App) versionedKeys(appVersion uint64) map[string]*storetypes.KVStoreKey {
+	output := make(map[string]*storetypes.KVStoreKey)
+	if keys, exists := app.keyVersions[appVersion]; exists {
+		for _, moduleName := range keys {
+			if key, exists := app.keys[moduleName]; exists {
+				output[moduleName] = key
+			}
+		}
+	}
+	return output
+}
+
+// baseKeys returns the base keys that are mounted to every version
+func (app *App) baseKeys() map[string]*storetypes.KVStoreKey {
+	return map[string]*storetypes.KVStoreKey{
+		// we need to know the app version to know what stores to mount
+		// thus the paramstore must always be a store that is mounted
```

Comment:
[not blocking][question] why is app version stored in the param store and consensus params? Is it stored anywhere else?

Reply:
No app version is only part of the consensus params in the param store. We load it in memory upon startup for easy reference.

I think the SDK ended up splitting out the param module which will likely be an annoying change to adopt